tests/kernel/common/src/clock.c

```diff
@@ -160,6 +160,35 @@ void test_clock_cycle(void)
 	}
 }
 
+/**
+ * @brief Test clock cycle 64 functionality
+ */
```

Comment:
This test really only checks to ensure that the 64-bit cycle counter grows proportionately to the 32-bit cycle counter on each platform.

I might add a sample application that periodically checks to ensure that the 64-bit timer has received updates to the upper 32-bits based on `CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC`.